devito/ir/iet/nodes.py

```diff
@@ -288,13 +288,16 @@ def is_tensor(self):
         return not self.is_scalar
 
     @property
-    def is_scalar_assign(self):
-        """True if a scalar, non-increment expression."""
-        return self.is_scalar and not self.is_Increment
+    def is_definition(self):
+        """
+        True if it is an assignment, False otherwise
+        """
+        return ((self.is_scalar and not self.is_Increment) or
+                (self.is_tensor and isinstance(self.expr.rhs, ListInitializer)))
 
     @property
     def defines(self):
-        return (self.write,) if self.is_scalar else ()
+        return (self.write,) if self.is_scalar or self.is_definition else ()
```

Comment:
I think the right thing to have here is ``self.is_definition`` , that is drop ``self.is_scalar or``

Reply:
i'm not sure if it has been decided already, but this sounds a little strange
i'd vote to something like `is_assign` or `is_assignment`

Reply:
@FabioLuporini I lest `self.is_scalar` because it's a different condition than the old `is_scalar_assign` (I wanted to avoid any change in behaviour due to that) but I agree it makes more sense to only have `is_definition`

@btrb a definition is a different thing compared to an assignment. Here we are specifically referring to a definition